azurerm/internal/services/monitor/monitor_smart_detector_alert_rule_resource.go

```diff
@@ -0,0 +1,328 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/monitor/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/monitor/validate"
+	azSchema "github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/set"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmMonitorSmartDetectorAlertRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmMonitorSmartDetectorAlertRuleCreateUpdate,
+		Read:   resourceArmMonitorSmartDetectorAlertRuleRead,
+		Update: resourceArmMonitorSmartDetectorAlertRuleCreateUpdate,
+		Delete: resourceArmMonitorSmartDetectorAlertRuleDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: azSchema.ValidateResourceIDPriorToImport(func(id string) error {
+			_, err := parse.SmartDetectorAlertRuleID(id)
+			return err
+		}),
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"detector_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"FailureAnomaliesDetector",
+				}, false),
+			},
+
+			"scope": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: azure.ValidateResourceID,
+				},
+				Set: set.HashStringIgnoreCase,
+			},
+
+			"severity": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice(
+					[]string{
+						string(alertsmanagement.Sev0),
+						string(alertsmanagement.Sev1),
+						string(alertsmanagement.Sev2),
+						string(alertsmanagement.Sev3),
+						string(alertsmanagement.Sev4),
+					}, false),
+			},
+
+			"frequency": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: commonValidate.ISO8601Duration,
+			},
+
+			"action_group": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ids": {
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Schema{
+								Type:         schema.TypeString,
+								ValidateFunc: validate.ActionGroupID,
+							},
+							Set: set.HashStringIgnoreCase,
+						},
+
+						"custom_email_subject": {
```

Comment:
custom seems redundant?
```suggestion
						"email_subject": {
```